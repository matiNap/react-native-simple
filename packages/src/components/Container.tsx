import React from "react";
import { StyleSheet, Text, View } from "react-native";

interface Props {
  children: React.ReactNode;
}

export default function Container({ children }: Props) {
  return <View style={styles.container}>{children}</View>;
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "black",
  },
});
