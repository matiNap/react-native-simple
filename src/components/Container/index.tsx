import React from 'react';
import { StyleSheet, View } from 'react-native';

import { getThemeColor, getProperty } from '../../helpers';
import useTheme from '../ThemeProvider/useTheme';
import Props from '../../types/Container';

export default ({ children, style, backgroundColor }: Props) => {
    const theme = useTheme();
    const { Container } = theme;

    return (
        <View
            style={[
                getProperty([Container?.style]),
                styles.container,
                {
                    backgroundColor: getProperty([
                        theme.currentPalette.secondary,
                        Container?.backgroundColor,
                        getThemeColor(backgroundColor, theme.currentPalette),
                    ]),
                },
                style,
            ]}
        >
            {children}
        </View>
    );
};

const styles = StyleSheet.create({
    container: { flex: 1 },
});
