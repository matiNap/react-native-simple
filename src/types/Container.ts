import { StyleProp, ViewStyle } from 'react-native';
import { SimpleBackgroundColor } from './index';

export interface LocalContainerProps {
    style?: StyleProp<ViewStyle>;
    backgroundColor?: SimpleBackgroundColor;
}

export default interface ContainerProps extends LocalContainerProps {
    children?: React.ReactNode;
}
