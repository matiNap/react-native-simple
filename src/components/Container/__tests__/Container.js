import React from 'react';
import { shallow } from 'enzyme';
import Container from '../index';
import toJson from 'enzyme-to-json';
import { describe, expect, it } from '@jest/globals';
import { Text } from 'react-native';

describe('Container component', () => {
  it('Should render', () => {
    const wrapper = shallow(<Container />);

    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should render with child', () => {
    const wrapper = shallow(
      <Container>
        <Text>Render with children</Text>
      </Container>,
    );

    expect(wrapper.length).toBe(1);

    expect(wrapper.children()).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should render with style', () => {
    const wrapper = shallow(
      <Container style={{ backgroundColor: 'black' }} />,
    );

    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
