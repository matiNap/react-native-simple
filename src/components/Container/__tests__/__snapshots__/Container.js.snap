// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component Should render 1`] = `
<View
  style={
    Array [
      null,
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#ededed",
      },
      undefined,
    ]
  }
/>
`;

exports[`Container component Should render with child 1`] = `
<View
  style={
    Array [
      null,
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#ededed",
      },
      undefined,
    ]
  }
>
  <Text>
    Render with children
  </Text>
</View>
`;

exports[`Container component Should render with style 1`] = `
<View
  style={
    Array [
      null,
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#ededed",
      },
      Object {
        "backgroundColor": "black",
      },
    ]
  }
/>
`;
